docs/howto/manage-metrics.md

```diff
@@ -0,0 +1,167 @@
+    secret = self.model.get_secret(id=secret_id)
+    content = secret.get_content()
+    username, password = content["username"], content["password"]
+    self._replace_identities(username, password)
+
+  def _on_secret_changed(self, event: ops.SecretChangedEvent) -> None:
+    if event.secret.id == self.config['metrics-secret-id']:
+      content = event.secret.peek_content()
+      username, password = content["username"], content["password"]
+      self._replace_identities(username, password)
+
+  def _replace_identities(self, username: str, password: str) -> None:
+    identities = {
+      username: ops.pebble.Identity(
+        access="metrics", basic=ops.pebble.BasicIdentity(password=sha512_crypt.hash(password))
+      ),
+    }
+    self.container.pebble.replace_identities(identities)
+    logger.debug("New metrics username: %s", username)
+
+  ...
+```
+
+The password of the Pebble identity is stored as a hash, which we generate using `sha512_crypt.hash()` from `passlib.hash`.
+
+When Pebble receives a request to access the metrics endpoint, Pebble will verify that the basic authentication credentials in the request match the identity's username and password.
+
+> See more:
+> - {ref}`make-your-charm-configurable`
+> - {ref}`manage-configurations`
+> - {ref}`manage-secrets`
+
+## Deploy the charm and grant access to the user secret
+
+We'll use a configuration file to set `metrics-secret-id` to the secret ID.
+
+First, create a configuration file named `metrics-config.yaml`:
+
+```yaml
+metrics-charm:
+  metrics-secret-id: <secret-id-here>
+```
+
+Then, when deploying the charm, use the `--config` option to pass the configuration file:
+
+```bash
+juju deploy <charm-name> --config metrics-config.yaml
+```
+
+After deploying the charm, grant access to the user secret:
+
+```bash
+juju grant-secret metrics-user-password <charm-name>
+```
+
+> See more:
+> - {external+juju:ref}`Juju | config <command-juju-config>`
+> - {external+juju:ref}`Juju | deploy <command-juju-deploy>`
+> - {external+juju:ref}`Juju | grant-secret <command-juju-grant-secret>`
+
+## Access the metrics endpoint
+
+### Within the same Kubernetes cluster
+
+Deploying the charm causes Juju to create a Kubernetes service named `<charm-name>-endpoints` within the Kubernetes cluster. Use this service to connect to Pebble within each workload container. Pebble's HTTP port for the first workload container is `38813`.
+
+For example, if you deploy a charm named `my-charm` in the `test` namespace, access the metrics endpoint at:
+
+```text
+my-charm-endpoints.test.svc.cluster.local:38813/v1/metrics
+```
+
+You'll need to use HTTP basic authentication with the username and password that you specified in the `juju add-secret` command.
+
+> See more: [Service discovery within a Kubernetes cluster](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/)
+
+### Through an Ingress
+
+To access the metrics endpoint from outside the Kubernetes cluster, use an Ingress.
```

Comment:
It might help with David's point above if we link the word Ingress. Can we link it to https://kubernetes.io/docs/concepts/services-networking/ingress/?